### PR TITLE
Remove error mapping in cosmology cloning

### PR DESCRIPTION
**Pull request by nstarman**
Signed-off-by: Nathaniel Starkman (@nstarman) <nstarkman@protonmail.com>

A deprecation marked for v5.0.
Request review: @mhvk

### CODE REVIEW

**Comment by pllim** [comment]
I thought we tend to avoid starting a deprecation in a LTS release, no?

**Comment by nstarman** [comment]
Sorry. I misspoke. This was already deprecated in v4.3 and is now removed.

File: docs/changes/cosmology/11785.api.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Deprecate warning and error remapping in ``Cosmology.clone``. Now unknown
```
Comment (pllim) [inline docs/changes/cosmology/11785.api.rst:1]: Change log needs fixing. Thanks!

Comment (nstarman) [reply docs/changes/cosmology/11785.api.rst]: Done. Thanks!

File: docs/changes/cosmology/11785.api.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Remove deprecated warning and error remapping in ``Cosmology.clone``.
+Now unknown arguments will raise a ``TypeError``, not a ``ValueError``.
```
Comment (pllim) [inline docs/changes/cosmology/11785.api.rst:2]:
```suggestion
Now unknown arguments will raise a ``TypeError``, not a ``AttributeError``.
```

**Review by mhvk — APPROVED** [review]
Great, thanks!

**Comment by mhvk** [comment]
But of course now it needs a rebase...

**Comment by mhvk** [comment]
OK, all approved, rebased and green, so merging! Thanks, @nstarman

**Pull request merged**